query-engine/connectors/mongodb-query-connector/src/query_builder/read_query_builder.rs

```diff
@@ -399,6 +409,27 @@ impl MongoReadQueryBuilder {
         Ok(self)
     }
 
+    /// Adds the necessary joins and the associated selections to the projection
+    pub fn with_aggregation_selections(
```

Comment:
Builders can be nice to abstract away some things, but in this particular instance I'm not sure you need it. Currently, the helper read builders are meant to encapsulate data that we need to process but can only do so at a later point because either:
- Some data from previous processing is required.
- We need to make a deferred decision during the actual query building.

However, you're just initializing the builder and then you're immediately consuming it.
Not only that, you have separate fields for the aggregation selections and joins, but why? I don't see you making deferred decision in the building for example, so the fields are unnecessary as far as I can tell - there's no reason you couldn't just add those directly to the normal joins or the projection.

Additionally, I'm not sure if this can produce double-joins. We may need to deduplicate the joins here.